dts/bindings/arm/nordic,nrf21540-gpio.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+
+description: >
+    This is a representation of the nRF21540 Radio Front-End module
+compatible: "nordic,nrf21540_gpio"
+
+include: base.yaml
+
+properties:
+    tx-en-gpios:
+        type: phandle-array
+        description: >
+            Gpio of the SOC controlling TX_EN pin of the nRF21540
+    tx-en-gpio-is-not-used:
```

Comment:
I think this is not necessary. You can make `tx-en-gpios` as optional properties by changing:
```
    tx-en-gpios:
        type: phandle-array
        required: false
        description: >
            Gpio of the SOC controlling TX_EN pin of the nRF21540
```
Then in application you can check by macros if DTS has this property.

Reply:
This file should be aligned with AdamZelikNS/sdk-nrf@d217b13. @AdamZelikNS, do you think this would be a better way to handle this configuration?

Reply:
The idea is that the user can choose not to control one of the pins (for example `tx-en`) by removing `tx-en-gpios` property, but must declare `tx-en-gpio-is-not-used` instead. This is to better visualize why `tx-en-gpios` property is missing in such case and to prevent possible mistakes

Reply:
With current proposal DTS would look like:

```
   / {
       nrf_radio_fem: name_of_fem_node {
           compatible = "nordic,nrf21540_gpio";
           tx-en-pin  = < &gpio0 13 ACTIVE_HIGH>;
           rx-en-gpio-is-not-used: true;
           pdn-pin    = < &gpio0 15 ACTIVE_HIGH>;
       };
   };
```

In your proposal it would be:
```
   / {
       nrf_radio_fem: name_of_fem_node {
           compatible = "nordic,nrf21540_gpio";
           tx-en-pin  = < &gpio0 13 ACTIVE_HIGH>;
           pdn-pin    = < &gpio0 15 ACTIVE_HIGH>;
       };
   };
```

It could end up with incorrect configurations created by mistake. In current binding, DTS creator has to explicitly state that given pin is not used, what should help avoiding mistakes in DTS creation.

Reply:
I agree with @KAGA164. In my humble opinion this is not necessary. You can make all `xxx-en-gpios` optional. If they are set as not required then end user just does not provide particular property. Less typing for end user and less code for driver maintainer. 

BTW. I've taken a quick look into documentation. Is there a use case that not all `tx_en`, `rx_en` and `pdn` are required to control the device?

Reply:
@hubertmis if we go that way, by stating explicitly that particular properties are disabled, wouldn't we end with forcing end users with explicit stating that e.g. peripherals are disabled?

Reply:
@ppryga , I don't have a strong opinion. I'm fine with both solutions. Just wanted to make sure that we understand the risk of mistakes if we make nRF21540 GPIO pins optional in DTS. As you noticed, these pins are __not__ optional in Product Specification.

@AdamZelikNS, @JakubPegza , what's your opinion?

Reply:
@hubertmis @ppryga I would be in favor of explicitly disabling control of specific pins, as this feels like less mistake-prone way, but I'm fine with both solutions.

Reply:
I would be in favor of removing the explicit disabling. It is not kind of a configuration that has internal-cross-dependencies, that we have to defend from being set inappropriately. I believe that if one starts to change it, he knows what he wants. Besides that, if pins are not optional, then I suggest to set `required=true` there. 

But I will not force it, let majority decides.  @KAGA164 ?

Reply:
> They could connect PDN to Vcc, or try to control TX and RX with single line and NOT logic.

How souch cases will be handled if for example PDN pin isn't physically connected to the board and pdn pin is mandatory so user still has to pass there some GPIO port and pin ? @hubertmis @JakubPegza